marin/resources_utils.py

```diff
@@ -0,0 +1,70 @@
+
+# https://github.com/stanford-crfm/levanter/blob/162ef5f321cfc2e14b1b9c1652e2cffd42b395e2/src/levanter/utils/flop_utils.py#L40
+
+import logging
+logger = logging.getLogger("ray")
+
+# Base FLOPS for each device type (using mixed precision values where available)
+device_flops_map = {
+    # GPUs (using mixed precision values where available)
+    "H100": 1.513e15 / 2,  # amp_fp16/bf16; pcie
+    "A100": 312e12,  # amp_fp16/bf16
+    "A10G": 70e12,  # bf16: https://www.baseten.co/blog/nvidia-a10-vs-a10g-for-ml-model-inference/
+    "V100": 125e12,  # amp_fp16
+    "T4": 65e12,  # amp_fp16
+    "A6000": 309.7e12 / 2,  # bf16/fp16
+    
+    # TPU base FLOPS per chip,
+    "v4": 275e12,  # bf16, https://cloud.google.com/tpu/docs/v4
+    "v5litepod": 197e12,  # bf16, https://cloud.google.com/tpu/docs/v5e
+    "v5": 459e12,  # bf16, https://cloud.google.com/tpu/docs/v5
+    "v6e": 918e12,  # bf16, https://cloud.google.com/tpu/docs/v6e
+}
+
+
+def get_tpu_type_and_chips(tpu_name: str) -> tuple[str, int]:
+    """Extract TPU type and number of chips from TPU name.
```

Comment:
give example of input and output

Actually, I just wonder if we should just declare all of this in a static map - there is only a small number of them and one can very easily look at the code to see what you'll get.  something like: `"v4-128": DeviceInfo(device_count=128*2, flops_per_sec=V4_FLOPS_PER_SEC)`

Reply:
(fwiw, it's `DeviceInfo(device_count=128 // 2, flops_per_sec=V4_FLOPS_PER_SEC)`, not `* 2`)

Reply:
and also... there are kind of a lot? it's not huge but big enough you'd want ot generate it programmatically. (~44 relevant configs currently)